### PR TITLE
Sp 14/testing calender events modification

### PR DESCRIPTION
**Pull request by guaj**
Fixed Sp-14 branch + e2e tests

### CODE REVIEW

**Review by mnashat — COMMENTED** [review]
One small change (check comment), but great job otherwise!

File: src/components/Calendar/Event/EditEvent.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+import { useCallback, useEffect } from "react";
+import { CalendarDatePicker, CalendarTextField, CalendarTimePicker, UpdateCancelButton, CompleteEditEvent } from '../Custom/CustomCalendarForms';
+
+export default function EditEvent() {
+    const [isRecurrent, setIsRecurrent] = React.useState(false);
+    const eventId = window.location.href.split("/")[window.location.href.split("/").length - 1];
+    const [eventData, setEventData] = React.useState({
+    })
+    const navigate = useNavigate();
+    const [eventError, seteventError] = React.useState({ message: "Error, please try again later", hasError: false });
```
Comment (mnashat) [inline src/components/Calendar/Event/EditEvent.js:21]: Should be `setEventError` for naming conventions.

File: src/components/Calendar/Calendar.js [hunk 2]
```diff
@@ -10,42 +10,56 @@ import { useNavigate } from "react-router";
...
 export default function CalendarView() {
 
-    const [date, setDate] = useState(new Date()) // stores date, sets date using Date obj
+    const [date, setDate] = useState(new Date()) 
 
-    const [event, setEvent] = useState([]);
+    const [events, setEvents] = useState([]);
+    const [acadEvents, setAcedemicEvents] = useState([]);
+
+    const [eventError, seteventError] = React.useState({ message: "Error, please try again later", hasError: false });
```
Comment (mnashat) [inline src/components/Calendar/Calendar.js:22]: What is the use of `eventError` state in this file? It is being set, but not used to display errors to the user (or anything else). Please remove this state if it's not being used.

**Review by mnashat — CHANGES_REQUESTED** [review]
Requested a few changes, great job otherwise!

File: src/components/Calendar/Event/EditEvent.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    const fetchData = useCallback(() => {
+        console.log("calling .get inside fetchData()")
```
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:24]: remove console.log

File: src/components/Calendar/Event/EditEvent.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+        axios.get(`${process.env.REACT_APP_BASE_URL}events/event/${eventId}`)
+            .then((res) => {
+                console.log(res)
```
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:27]: remove console.log

File: src/components/Calendar/Event/EditEvent.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+                setEventData(
+                    res.data
+                )
+            }
+            ).catch((err) => {
+                seteventError({ ...eventError, message: err.message });
+                console.log(err)
```
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:34]: remove console.log

File: src/components/Calendar/Event/EditEvent.js [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+            });
+    }, [])
+    useEffect(() => {
+        console.log("Calling fetchData()")
```
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:38]: remove console.log

File: src/components/Calendar/Event/EditEvent.js [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+        </FormControl>
+    );
+
+    function handleReccurenceChange(e) {
+        setEventData({ ...eventData, reccurence: e.target.value })
+    }
+
+    function handleEvent() {
+        // TODO:  validate user inputs if have time
+        console.log(eventData)
```
Comment (kikkiii21) [inline src/components/Calendar/Event/EditEvent.js:72]: remove console.log

File: src/components/FriendList/FriendSearch.js [hunk 8]
```diff
@@ -64,8 +67,15 @@ export default function FriendSearch() {
                                     if (isInFriends) {
                                         setText("already a friend");
                                     } else {
-                                        setFriend(res.data.email);
-                                        setFound(true);}
+                                        if (res.data.email) {
+                                            console.log(res.data.email);
```
Comment (kikkiii21) [inline src/components/FriendList/FriendSearch.js:71]: remove console.log

File: cypress/support/friends.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Navigate to the friend list page with the menu.
+ */
+Cypress.Commands.add('goToFriendList', () => {
+    cy.get('[data-test="navbar"]').click();
+    cy.get('[data-test="Friends List"]').click();
```
[the remark this answers is not in the document]
Comment (guaj) [reply cypress/support/friends.js:6]: yes

**Review by c0rnkrump3t — APPROVED** [review]

File: cypress/e2e/friends.cy.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            true
+        );
+        cy.changeUser('email1', 'password1');
+        cy.goToFriendList();
+        cy.get('[data-test="friendRequestsLink"]').click();
+        cy.get(`[data-test="accept-request-${Cypress.env('email2')}"]`).click();
+    })
+
+    it('User can delete a friend', function () {
+        cy.get(`[data-test="delete-friend-${Cypress.env('email1')}"]`).click();
```
Comment (based-kwl) [inline cypress/e2e/friends.cy.js:49]: This test also sends duplicate requests, but are not caught by the BE and remain valid.

File: cypress/e2e/friends.cy.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+    it('User search for a non existing user and tries to add him', function() {
+        cy.addUser(
+            randomEmail,
+            false
+        );
+    })
+
+    it('User can cancel a friend request', function() {
+        cy.cancelRequest(Cypress.env('email1'));
```
Comment (based-kwl) [inline cypress/e2e/friends.cy.js:34]: This test sends a duplicate request.

**Review by based-kwl — CHANGES_REQUESTED** [review]
Some minor issues with the e2e testing, otherwise good job.

File: src/components/FriendList/FriendsNotification.js [hunk 12]
```diff
@@ -22,6 +22,7 @@ export default function FriendNotification() {
         const id = requestSent[index]._id;
         axios.post(`${process.env.REACT_APP_BASE_URL}friend/cancel-request`,{requestId:id})
```
Comment (based-kwl) [inline src/components/FriendList/FriendsNotification.js:23]: include the email in the request as `email`

**Review by based-kwl — CHANGES_REQUESTED** [review]

**Comment by guaj** [comment]
Last test result : 
image

**Review by based-kwl — APPROVED** [review]

File: src/components/Calendar/Event/EditEvent.js [hunk 13]
```diff
@@ -117,11 +103,8 @@ export default function EditEvent() {
 
     const editUpdateButtons = (
         <React.Fragment>
-          
-    
             <UpdateCancelButton backgroundColor={'#912338'} content="Update" onClick={() => { handleEvent() }} />
             <UpdateCancelButton backgroundColor={'#C8C8C8'} content="Cancel" onClick={() => { handleEvent() }} />
```
Comment (mnashat) [inline src/components/Calendar/Event/EditEvent.js:107]: Why does the cancel button call handleEvent()? It should not make a post request to the database.

**Review by mnashat — APPROVED** [review]

File: src/components/Calendar/Calendar.js [hunk 14]
```diff
@@ -10,43 +10,55 @@ import { useNavigate } from "react-router";
...
 import axios from "axios";
 
 export default function CalendarView() {
 
-    const [date, setDate] = useState(new Date()) // stores date, sets date using Date obj
+    const [date, setDate] = useState(new Date()) 
 
-    const [event, setEvent] = useState([]);
+    const [events, setEvents] = useState([]);
+    const [academicEvents, setAcedemicEvents] = useState([]);
```
Comment (kikkiii21) [inline src/components/Calendar/Calendar.js:20]: there's a typo, it should be setAcademic not setAcedemic

**Review by kikkiii21 — APPROVED** [review]

**Pull request merged**